### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by TheComputerGuy96**
Check each commit for more details

### CODE REVIEW

**Comment by ghost** [comment]
Thank you for the PR
- Removing LatinIME patch is not necessary because it's just a fallback (just in case) if there is other GAPPS did not fix this so I think I will still keep this in patch, there is no harm.
- I did not tested the config_enableAutoPowerModes so don't know if it's working and/or causing any other problem so I'll merge this after test to make sure things won't broke.

I'll test and cherry-pick from your PR as soon as possible.

**Comment by TheComputerGuy96** [comment]
We need to modify the LatinIME patch so it points only to working gesture typing lib which we will add in vendor repo

**Comment by TheComputerGuy96** [comment]
Thanks, SandPox

**Comment by TheComputerGuy96** [comment]
https://github.com/CyanogenMod/android_frameworks_base/blob/cm-13.0/location/java/android/location/GpsMeasurement.java#L63

**Comment by TheComputerGuy96** [comment]
I closed this by a accident

**Comment by TheComputerGuy96** [comment]
Waiting for the testing of these commits

File: rootdir/init.hawaii_ss_kylepro.rc [hunk 1]
```diff
@@ -608,6 +608,7 @@ on property:init.svc.macloader=stopped
 # End of wifi
 
     service gpsd /system/bin/glgps -c /system/etc/gps/glconfig.xml
+        setenv LD_PRELOAD /system/lib/libglgps-compat.so
```
Comment (ghost) [inline rootdir/init.hawaii_ss_kylepro.rc:611]: gpsd load glgps-compat is fine because there is no missing lib error in log

**Comment by TheComputerGuy96** [comment]
Well, there is (probably):
https://github.com/TheComputerGuy96/random_crap/tree/master/kmesg.txt#L6

**Comment by TheComputerGuy96** [comment]
And another ones:
http://imgur.com/a/Kngla

**Comment by ghost** [comment]
^The GPS problem is because of missing symbol. I'll fix this when I have time

**Comment by ghost** [comment]
Using Snap-Camera is not a good idea, snap camera focuses Qualcomm most. I tested that and had force close when changing to front cam.

**Comment by TheComputerGuy96** [comment]
See this, ishantvivek;
https://github.com/SandPox/android_device_samsung_kylepro/pull/3/commits/392adf4e3096a4399c451842eb0e6639306dc74e

**Comment by ghost** [comment]
Hello
I just tested your PR, here is the result:
- Auto power mode, remove deprecated flag will be implement
- Fix compilation warning for libhealth will NOT merge because it's don't make any sense
- LatinIME related commit will NOT be implement because firstly it's a fallback, does no harm; OpenGAPPS fix that don't mean every other GAPPS fixed that. Secondly you said that you remove this patch to use specify keyboard lib, I'm not agree with that because I want to minimize the use of prebuilt binary as much as possible and the lib might broke the keyboard if keyboard application get updated but the lib is static binary.
- Screen recording fix is not working so will not be implement.
- Snap camera working pretty well, it will be implement.
- remove persist.sys.usb.config override will be implement.
- Fix GPS is not working because it's not caused by that lib but by missing a symbol, I'll fix this in the future.

Your commit is pretty messy so I'll pick them rewrite the title, message to keep things organized but of course it will still get your author name.

Well done DODOGTA, if you finding such useful things then surely I'll use them, just better to check if it's work next time and the cons/pros of it if it get implemented before commit and making pull request

**Comment by minhng99** [comment]
Picked and merged into cm-13.0 branch.

**Pull request closed without merging**